ostd/src/mm/page_table/node/mod.rs

```diff
@@ -152,135 +125,83 @@ impl<E: PageTableEntryTrait, C: PagingConstsTrait> RawPageTableNode<E, C> {
     pub(super) unsafe fn first_activate(&self) {
         use crate::{arch::mm::activate_page_table, mm::CachePolicy};
 
-        self.inc_ref_count();
-
-        activate_page_table(self.raw, CachePolicy::Writeback);
+        activate_page_table(self.clone().into_raw(), CachePolicy::Writeback);
     }
+}
 
-    fn inc_ref_count(&self) {
-        // SAFETY: We have a reference count to the page and can safely increase the reference
-        // count by one more.
-        unsafe {
-            inc_frame_ref_count(self.paddr());
-        }
-    }
+/// A reference to a page table node.
+pub(super) type PageTableNodeRef<'a, E, C> = FrameRef<'a, PageTablePageMeta<E, C>>;
 
-    /// Restores the handle from the physical address and level.
-    ///
-    /// # Safety
+impl<'a, E: PageTableEntryTrait, C: PagingConstsTrait> PageTableNodeRef<'a, E, C> {
+    /// Locks the page table node.
     ///
-    /// The caller must ensure that the physical address is valid and points to
-    /// a forgotten page table node. A forgotten page table node can only be
-    /// restored once. The level must match the level of the page table node.
-    pub(super) unsafe fn from_raw_parts(paddr: Paddr, level: PagingLevel) -> Self {
-        Self {
-            raw: paddr,
-            level,
-            _phantom: PhantomData,
+    /// An atomic mode guard is required to prevent deadlocks.
```

Comment:
"to (1) prevent deadlocks and (2) ensure that all stored nodes are not deallocated before the RCU grace period."